rero_ils/modules/items/dumpers.py

```diff
@@ -42,3 +44,28 @@ def dump(self, record, data):
         }
         data = {k: v for k, v in data.items() if v}
         return data
+
+
+class ItemCirculationDumper(InvenioRecordsDumper):
+    """Item dumper class for circulation."""
+
+    def dump(self, record, data):
+        """Dump an item instance for circulation.
+
+        :param record: the record to dump.
+        :param data: the initial dump data passed in by ``record.dumps()``.
+        :return a dict with dumped data.
+        """
+        # Dump all information about the item
+        data.update(record.replace_refs().dumps())
+        data = {k: v for k, v in data.items() if v}
+
+        # Add the inherited call numbers from parent holding record if item
+        # call numbers is empty.
+        if all(k not in data for k in ['call_number', 'second_call_number']):
+            holding = Holding.get_record_by_pid(record.holding_pid)
+            data['call_number'] = holding.get('call_number')
+            data['second_call_number'] = holding.get('second_call_number')
+            data = {k: v for k, v in data.items() if v}
```

Comment:
maybe you can only do `data = {k: v for k, v in data.items() if v}` at the end of method

Reply:
no, because, it's possible than the item "second_call_number" exists but was empty. In this case, the key exists and inherited holding call numbers isn't set.